ledger/block/src/transactions/confirmed/mod.rs

```diff
@@ -79,10 +79,11 @@ impl<N: Network> ConfirmedTransaction<N> {
                     program.mappings().len(),
                 )
             }
-            // Ensure the number of finalize operations matches the number of 'UpdateKeyValue' finalize operations.
-            if num_update_key_values != fee.num_finalize_operations() {
+            // Ensure the number of fee finalize operations lower bounds the number of 'UpdateKeyValue' finalize operations.
+            // The lower bound is due to the fact that constructors can issue 'UpdateKeyValue' operations as part of the deployment.
+            if num_update_key_values < fee.num_finalize_operations() {
```

Comment:
Is there a way to fetch the number of finalize operations from the program constructor? 

I realize that `num_finalize_operations` is a function that only exists for fees currently and is hardcoded to 1 or 0.

Reply:
You could get an upper bound.
You can't get an exact number since a constructor may contain branches.